.github/workflows/tests.yml

```
@@ -69,6 +70,13 @@ jobs:
           max_attempts: 5
           command: composer require guzzlehttp/psr7:^2.4 --no-interaction --no-update
 
+      - name: Set PHPUnit
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 5
+          max_attempts: 5
+          command: composer require phpunit/phpunit:~${{ matrix.phpunit }} --dev --no-interaction --no-update
```

```suggestion
          command: composer require phpunit/phpunit:^${{ matrix.phpunit }} --dev --no-interaction --no-update
```

the tilde is only necessary for Windows.